astropy/time/core.py

```diff
@@ -1412,6 +1413,68 @@ def value(self):
         return (self.jd1 - erfa.DJM0) + self.jd2
 
 
+class TimeDecimalYear(TimeFormat):
+    """
+    Time as a decimal year, with integer values corresponding to midnight
+    of the first day of each year.  For example 2005.5 is corresponds to the
+    ISO time '2000-07-02 00:00:00'.
+    """
+    name = 'decimalyear'
+
+    def set_jds(self, val1, val2):
+        self._check_scale(self._scale)  # Validate scale.
+
+        val = (val1 + val2).astype(np.double)
+        print(val)
+        iy_start = np.trunc(val).astype(np.int)
```

Comment:
We can do the split in years and fraction to somewhat better precision with a short version of `day_frac`:
```
sum12, err12 = two_sum(val1, val2)
iy_start = np.trunc(sum12).astype(np.int)
extra, y_frac = two_sum(sum12, -iy_start)
y_frac += extra + err12 
```